src/pages/workspace/WorkspacesListPage.js

```diff
@@ -131,7 +131,11 @@ class WorkspacesListPage extends Component {
                 action: () => Navigation.navigate(ROUTES.getWorkspaceInitialRoute(policy.id)),
                 iconFill: themeColors.textLight,
                 fallbackIcon: Expensicons.FallbackWorkspaceAvatar,
-                brickRoadIndicator: reimbursementAccountBrickRoadIndicator || PolicyUtils.getPolicyBrickRoadIndicatorStatus(policy, this.props.policyMembers),
+                brickRoadIndicator:
+                    PolicyUtils.hasPolicyErrorFields(policy)
+                    || reimbursementAccountBrickRoadIndicator
+                    || PolicyUtils.getPolicyBrickRoadIndicatorStatus(policy, this.props.policyMembers)
+                        ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : null,
```

Comment:
Hmmm...this seems a bit wonky.

The `reimbursementAccountBrickRoadIndicator` and `PolicyUtils.getPolicyBrickRoadIndicatorStatus` either return `CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR` or `''`, while `PolicyUtils.hasPolicyErrorFields` returns a boolean.

I think it would be better to redefine the `reimbursementAccountBrickRoadIndicator` to be...

```js
const reimbursementAccountOrPolicyBrickRoadIndicator = (!_.isEmpty(this.props.reimbursementAccount.errors) || PolicyUtils.hasPolicyErrorFields(policy)) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : '';
```

Then, for this parameter, we would have:
```js
brickRoadIndicator: reimbursementAccountOrPolicyBrickRoadIndicator || PolicyUtils.getPolicyBrickRoadIndicatorStatus(policy, this.props.policyMembers);
```

Reply:
Actually policy belongs to `policies` list thus must be traversed through `map` function; while `reimbursementAccount` is the same for every policy. Thus I don't think we can refactor as your suggestion.

Reply:
For the inconsistent return value, you can refer the same usage here:

https://github.com/Expensify/App/blob/32fbb2c44aeea0e112f8eb0ca9ea4911e160f499/src/pages/settings/InitialSettingsPage.js#L173-L174

However, I also think this is not good though I can't think of another way to make this logic cleaner. Want to here from you!

Reply:
Okay so I found a solution for this. We need to make `getPoliciyBrickRoadIndicator` function to include the `hasPolicyErrorFields` condition too. I think this make the function more "meaningful".

https://github.com/Expensify/App/blob/32fbb2c44aeea0e112f8eb0ca9ea4911e160f499/src/libs/PolicyUtils.js#L52

to

```javascript
if (hasPolicyMemberError(policyMemberList) || hasCustomUnitsError(policy) || hasPolicyErrorFields(policy))
```